docs/design/rosbag2_record_replay_service.md

```diff
@@ -0,0 +1,118 @@
+# Rosbag2 record and replay service
+
+## Context
+
+In [ros2/rosbag2#773](https://github.com/ros2/rosbag2/issues/773), it was recommended that record service requests feature should be added to rosbag2.  
+
+This design describe how to implement this feature based on current architecture and how to extend current command parameter.  
+
+## How to save record service message
+
+Service introspection ([ros2/ros2#1285](https://github.com/ros2/ros2/issues/1285)) had been implemented. All service messages (requests and responses) are published to service event topic which is hidden topic named as `_service_event` under service name (e.g. /add_two_ints/_service_event). So all messages in this topic can be recorded like general topic.
+
+## How to distinguish service event topic in all recorded topics
+
+Service event topic is saved like general topic. But they should use different process after reading from recording file. So they should be distinguished by topic name.
+
+- General topic  
+  
+  [NameSpace/]TopicName
+
+- Hidden topic  
+  
+  [NameSpace/]_TopicName
+
+- Service event topic
+  
+  [NameSpace/]ServiceName/_service_event
+
+So if the string ends with `/_service_event` in topic name, it should be service event topic. Of course, it is possible that hidden topic has the same name. So message type is used for further check. The message type for service event topic must include `/srv/` and ends with `_Event` (e.g. example_interfaces/srv/AddTwoInts_Event). Therefore, service event topic can be identified.
+
+### Expand the 'record' command
+
+Add 3 parameters.
+
+- `-S SerivceName1 [ServiceName2 ...]` or `--services SerivceName1 [ServiceName2 ...]`
+  
+    If the user doesn't specify service names, all service event topics are recorded.
```

Comment:
This is overloaded with `--all-services`. I would prefer that `-S,--services` takes `nargs='+'` so the user _must_ specify some service if they use this option.

Reply:
I forgot to update this description. User must specify at least one service.